src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamAccessService.java

```diff
@@ -47,30 +47,26 @@ public StudentExamAccessService(CourseRepository courseRepository, UserRepositor
      * @param courseId      the if of the course
      * @param examId        the id of the exam
      * @param studentExamId the id of the student exam
-     * @param isTestRun     flag to determine if it is a test run or not
      */
-    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, Long studentExamId, boolean isTestRun) {
-        checkStudentExamAccessElseThrow(courseId, examId, studentExamId, userRepository.getUserWithGroupsAndAuthorities(), isTestRun);
+    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, Long studentExamId) {
+        StudentExam studentExam = studentExamRepository.findByIdElseThrow(studentExamId);
+        checkStudentExamAccessElseThrow(courseId, examId, studentExam, userRepository.getUserWithGroupsAndAuthorities());
     }
 
     /**
      * Checks if the current user is allowed to see the requested student exam.
      *
      * @param courseId      the if of the course
      * @param examId        the id of the exam
-     * @param studentExamId the id of the student exam
+     * @param studentExam   the student exam
```

Comment:
I would add that this entity should not be sent by the client but needs to be fetched from the database, otherwise this could impose a security issue

Reply:
✅  325db4f